pkg/engine/jsonContext.go

```diff
@@ -53,7 +53,7 @@ func LoadContext(ctx context.Context, logger logr.Logger, rclient registryclient
 
 		if rule != nil && len(rule.ForEachValues) > 0 {
 			for key, value := range rule.ForEachValues {
-				if err := enginectx.jsonContext.AddVariable(key, value[store.ForeachElement]); err != nil {
+				if err := enginectx.jsonContext.AddVariable(key, value[0]); err != nil {
```

Comment:
@vyankyGH You initially added this functionality. While trying to fix a TODO comment, I tried removing the logic completely and just always taking the first array element. It seems like none of the tests fail though.

Do you recall why the is increased like it is? For some reason it's increased on every mutate / validate call. This shouldn't have worked to begin with in policies with multiple calls, right?
And also won't work with nested foreach now?

I am just quite confused how this was intended to work.